src/http_parser.cpp

```diff
@@ -209,7 +210,14 @@ namespace libtorrent {
 			m_protocol = read_until(line, ' ', line_end);
 			if (m_protocol.substr(0, 5) == "HTTP/")
 			{
-				m_status_code = atoi(read_until(line, ' ', line_end).c_str());
+                char *str;
+				m_status_code = strtol(read_until(line, ' ', line_end).c_str(), &str, 10);
+			    if (*str)
```

Comment:
the tests are failing, and I'm suspecting that this assumption that `str` won't be set to anything isn't correct. there may be white space or a newline following the number. It's not an error.

Reply:
I would suggest wrapping the use of `strtol()` in a function (probably in `string_utils.cpp`) that does all the error handling and has a simpler interface.

In fact, it may even be simpler to implement the string to number conversion "manually" than to handle all the error cases from the standard functions.

Reply:
mm, i think first we/i (maybe you already know this) need know what format we should support, how there was trying to get numbers i suppose the strings would be that, but if not i think is necessary define the format to avoid weird things.

Reply:
HTTP is fairly well defined

Reply:
Hi, in that case we should be getting just a number, maybe we just need check what we are getting for the error.

Reply:
`strtol()` is documented here: https://en.cppreference.com/w/cpp/string/byte/strtol
"The functions sets the pointer pointed to by str_end to point to the character past the last character interpreted".

Your check `if (*str)` does *not* indicate an error, you're just checking whether there *is* a character after the number

Reply:
right, the idea is check if there is something more than a number, if there is something more i think we should read until that chracter and from there get the number.

Reply:
I'm supposing there's white space *after* the number, which is perfectly fine in HTTP.